src/basic/namespace-util.c

```diff
@@ -47,6 +47,28 @@ static NamespaceType clone_flag_to_namespace_type(unsigned long clone_flag) {
         return _NAMESPACE_TYPE_INVALID;
 }
 
+static int pidref_namespace_open_by_type(const PidRef *pidref, NamespaceType type) {
```

Comment:
So I'm already working on a bigger rework of this, in order to support PIDFD_GET_*_NS. Could you drop the last commit?

Reply:
You mean #35242 ? I cannot find anything conflicting ?
But OK. No problem. I keep the last commit for a while, but will drop when your PR is ready.

Reply:
Or, feel free to pick the first two into your PR if you'd like.
Anyway, this will not be merged so soon. We hopefully have enough time.